.github/workflows/test.yml

```
@@ -86,18 +79,100 @@ jobs:
-          CHANGES=$(git status -s)
-          if [[ ! -z $CHANGES ]]; then
-            echo "Changes found: $CHANGES"
-            exit 1
+          gh pr view ${{ github.event.pull_request.number }} --json files -q '.files[] | select (.status != "removed") | .path' > changed_files.txt
+          echo "Changed files: $(cat changed_files.txt)"
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Run prettier (changed files)
+        if: github.event_name == 'pull_request'
+        run: >-
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2002:style:1:5: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead

(shellcheck)